### ISSUE TITLE
Move h-api out into a separate library for use in LMS

### ISSUE THREAD

**Issue by jon-betts** [hypothesis/h#6004]
Now that the code is (hopefully) more likely to be stable we can move it out into a separate library. This is to enable us to re-use the code two write the client portion in LMS.

At the end of this ticket we should have:

 * No h-api code left in h
 * A separate library on pypi with the new code
 * Which builds and tests itself using Github actions with all the cookie cutter goodness

Tasks:

 * [x] Create blank cookie cutter (https://github.com/hypothesis/h-api)
 * [x] Copy code verbatim (+ renaming h.h_api -> h_api) (https://github.com/hypothesis/h-api/pull/1)
 * [x] Fix the code to actually work as a package
 * [x] Deploy to PyPI (...eugh... I've messed this up)

**Comment by jon-betts** [comment]
So I have uploaded the package to PyPI, but the way I've been doing this is to upload them as myself, then assign ownership to the eng account and generate a key for github.

Unfortunately I seem to have lost my 2FA for my personal account. I could swear I never set it up (so yay me for being more secure?). Anyway, I can still update the package manually, but nobody else can. So technically this isn't blocked for now.

I've started the account recovery process and we'll see what happens. Worst comes to worst, I can rename and upload it as "h-api-client" or something using a generic key on the eng account. I didn't want to use a generic API key on the main account as it's unnecessarily powerful most of the time.
